clueless-client/src/main/java/edu/jhu/espresso/client/MessageStub.java

```diff
@@ -0,0 +1,78 @@
+package edu.jhu.espresso.client;
+
+import java.util.List;
+
+public final class MessageStub implements Comparable<MessageStub>
+{
+    private List<String> validMoves;
+    private TurnIndicator turnIndicator;
+    private int handlerNumber;
```

Comment:
These fields aren't immutable; That's so jackson can set the values with the appropriate setter methods. One way to use jackson deserialize into an immutable object is to use a [builder](https://blogs.oracle.com/javamagazine/post/exploring-joshua-blochs-builder-design-pattern-in-java).